### PR TITLE
[DDW-575] Handle 'transaction is too big' transaction error

### PR DESCRIPTION
**Pull request by daniloprates**
This PR implements a handle for the 'transaction is too big' error in the "fee calculation:" and "create transaction" actions.

This error happens the user making the transaction has their amount distributed among too many addresses, therefore the transaction needs to use too many input addresses. 

## Todo:

- [x] Implement the error handle for the `TooBigTransaction` error message
- [x] Final English copy: `Amount too big due to wallet fragmentation. Learn more.` - @darko-mijic 
- [x] Japanese translation: `ウォレットのフラグメンテーションの為、額が大きすぎます。もっと詳しく知る` - @nikolaglumac 
- [x] Replace all mentions of `Ada` with `ada` - it should NOT be written with a capital A - @daniloprates 

*Note*: The text part `Learn more` should be a link to:
- English article: https://iohk.zendesk.com/hc/en-us/articles/360017733353
- Japanese article:  https://iohk.zendesk.com/hc/ja/articles/360017733353

## Screenshot:

screen shot 2019-02-11 at 13 09 09

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

File: source/renderer/app/api/api.js [hunk 1]
```diff
@@ -456,6 +457,9 @@ export default class AdaApi {
           throw new NotEnoughFundsForTransactionError();
         }
       }
+      if (error.message === 'TooBigTransaction') {
```
Comment (nikolaglumac) [inline source/renderer/app/api/api.js]: @daniloprates please add this error handling to `createTransaction` too 🙏

Comment (daniloprates) [reply source/renderer/app/api/api.js]: Ok

Comment (daniloprates) [reply source/renderer/app/api/api.js]: ✅

**Comment by nikolaglumac** [comment]
@daniloprates here's Japanese translation:
```
Amount too big due to wallet fragmentation. Learn more.
```
```
ウォレットのフラグメンテーションの為、額が大きすぎます。もっと詳しく知る
```
...the last part `もっと詳しく知る` should be linked.

**Comment by daniloprates** [comment]
@DominikGuzei @MarcusHurney for this task it was implemented a way to add links to a [`LocalizableError`](https://github.com/input-output-hk/daedalus/pull/1308/files#diff-0d606537fcfe89090ad23936130f708dL1). This is how it was done:

* In the [`TooBigTransactionError`](https://github.com/input-output-hk/daedalus/pull/1308/files#diff-621b21e1bbd930e754e04b0ed1ef92eaR55) error,  the [`linkLabel`](https://github.com/input-output-hk/daedalus/pull/1308/files#diff-621b21e1bbd930e754e04b0ed1ef92eaR50) and [`linkURL`](https://github.com/input-output-hk/daedalus/pull/1308/files#diff-621b21e1bbd930e754e04b0ed1ef92eaR55) intl messages [are sent](https://github.com/input-output-hk/daedalus/pull/1308/files#diff-621b21e1bbd930e754e04b0ed1ef92eaR140) via the `values` object
* In the places where the error is displayed [it checks](https://github.com/input-output-hk/daedalus/pull/1308/files#diff-cab11d6f15fbdbb099e874083dd00bc1R168) if there is the `error.values.linkLabel` variable
* If so, [it](https://github.com/input-output-hk/daedalus/pull/1308/files#diff-cab11d6f15fbdbb099e874083dd00bc1R171) [uses](https://github.com/input-output-hk/daedalus/pull/1308/files#diff-cab11d6f15fbdbb099e874083dd00bc1R171) a new class called [`FormattedHTMLMessageWithLink`](https://github.com/input-output-hk/daedalus/pull/1308/files#diff-8d16bfac011f790d9bb3c3c56d33bbc3R1), which facilitates rendering translated text with links.

**Comment by daniloprates** [comment]
For the "_Replace all mentions of `Ada` with `ada`_" part, this part should remain with uppercase letters:

image

**Comment by DominikGuzei** [comment]
@daniloprates can you tell us an easy way how to test this error?

**Comment by daniloprates** [comment]
> source/renderer/app/components/widgets

@MarcusHurney makes total sense. [Done](https://github.com/input-output-hk/daedalus/pull/1308/commits/5ecb74467e300e7739ce263c5be62acaa9f7c1c5).

**Comment by daniloprates** [comment]
> @daniloprates can you tell us an easy way how to test this error?

To test it in the fee calculation, replace the [Api](https://github.com/input-output-hk/daedalus/blob/5ecb74467e300e7739ce263c5be62acaa9f7c1c5/source/renderer/app/api/api.js#L410) lines 410 - 466 by `throw new TooBigTransactionError();`

For the transaction, replace the lines 362 - 399.

**Comment by MarcusHurney** [comment]
@daniloprates 
I pushed a commit fixing the import path that was off after moving the `FormattedHTMLMessageWithLink.js` component to the `/widgets` folder.

Outside of that little fix, the error message looks like it is working correctly when I tested it in `calculateTransactionFee` and `createTransaction`. The only flaw I found is in the styling of the error message (with a link) when it is rendered in Japanese within the [WalletSendConfirmationDialog.js](https://github.com/input-output-hk/daedalus/blob/5ecb74467e/source/renderer/app/components/wallet/WalletSendConfirmationDialog.js#L171)
The text is longer in Kanji than in English, so the link wraps to the next line. Here is a screenshot:

screen shot 2019-02-13 at 10 55 47 pm

In `WalletSendForm.js` the styling is fine in Japanese 👌:

screen shot 2019-02-13 at 10 57 23 pm

**Comment by darko-mijic** [comment]
@MarcusHurney, we need DDW-565 completed for 0.13.0. Maybe we should wait while both this PR and that one are merged before creating the release branch.

**Comment by DominikGuzei** [comment]
@daniloprates @MarcusHurney im wondering why the error should be shown on the confirmation dialog? Shouldn't the submit button be disabled if the error is shown on the form (and thus the user would never get to the confirmation dialog?)

**Comment by daniloprates** [comment]
image

I could reproduce the issue in the calculation part. But there's not much to do, here, from our side. I've tried `letter-spacing: -.7px`, but it looked awful in English. Maybe ask the Japanese translator a shorter sentence or open a card to improve the error position and accommodate longer messages.

**Comment by nikolaglumac** [comment]
@daniloprates @DominikGuzei @MarcusHurney is this one ready to be merged?

**Comment by DominikGuzei** [comment]
@nikolaglumac yes i think so, the PR itself works correctly - the only thing that Danilo did since last week was to fix the ADA renaming stuff … not sure how critical those updates are but it seems like his last three commits have been dedicated to iron out the last issues.

**Comment by nikolaglumac** [comment]
Thanks @DominikGuzei - I will review it again.

**Pull request merged**